classes/class-blog-handler.php

```diff
@@ -84,21 +102,44 @@ public function init( array $data, $user = null ) {
 	 * @return int
 	 */
 	public function first_or_create_blog( $make_private = false ) {
+
+		$this->validate_blog_data();
+
+		$blog_id = $this->get_blog_id_if_exists();
+
+		if ( null === $blog_id ) {
+			$this->fix_blog_id_from_path();
+			$blog_id = $this->get_blog_id_if_exists();
+		}
+
+		if ( null === $blog_id ) {
+			$blog_id = $this->create_blog( $make_private );
+		}
+
+		$this->blog_id = $blog_id;
+
+		return $blog_id;
+	}
+
+	/**
+	 * Manage additional blog options when a blog is created or viewed
+	 *
+	 * @return string
+	 */
+	public function set_additional_blog_options( array $options_to_set ) {
+		foreach ( $options_to_set as $blog_option_key => $blog_option_value ) {
+			return update_blog_option( $this->blog_id, $blog_option_key, $blog_option_value );
```

Comment:
The function says we  will return a string but there is a chance we could return void if no options_to_set are passed. I doubt this will actually happy in practice but would be good to match the `@return` statement accordingly.

Also wondering if the `foreach` is necessary if we are returning on the first item?

Reply:
Thanks for spotting, I think I changed this function while working on it, and didn't pick up on the above.

The return is not required, it should iterate and set any blog options sent.

I have updated.

Reply:
LGTM. Cheers Dave